src/MultiMap.ts

```diff
@@ -0,0 +1,218 @@
+import Symbol from '@dojo/shim/Symbol';
+import Map from '@dojo/shim/Map';
+import { ArrayLike } from '@dojo/shim/interfaces';
+import { forOf, Iterable, IterableIterator, ShimIterator } from '@dojo/shim/iterator';
+
+/**
+ * A map implmentation that supports multiple keys for specific value.
+ *
+ * @param T Accepts the type of the value
+ */
+export default class MultiMap<T> {
```

Comment:
A good question, I probably should `implement` rather than `extend` `Map` (and doing so means I need to add `size`)